pkg/controller/node.go

```diff
@@ -38,29 +38,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-// ReconcileAzDriverNode reconciles AzDriverNode
-type ReconcileAzDriverNode struct {
+// ReconcileNode reconciles AzDriverNode
+type ReconcileNode struct {
 	*SharedState
 	logger logr.Logger
 }
 
 // Implement reconcile.Reconciler so the controller can reconcile objects
-var _ reconcile.Reconciler = &ReconcileAzDriverNode{}
+var _ reconcile.Reconciler = &ReconcileNode{}
 
-func (r *ReconcileAzDriverNode) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+func (r *ReconcileNode) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
 	if !r.isRecoveryComplete() {
 		return reconcile.Result{Requeue: true}, nil
 	}
+	logger := r.logger.WithValues(consts.NodeNameLabel, request.Name)
 
 	n := &corev1.Node{}
 	err := r.cachedClient.Get(ctx, request.NamespacedName, n)
 
 	if err == nil {
```

Comment:
np: could we reorder and check in the following order so that we don't have to reconcile in the "default" case: 

1. errors.IsNotFound(err)  ->  ...delete azdrivernode...
2. error != nil -> reconcile.Result{Requeue: true}, err
3. if n.ObjectMeta.DeletionTimestamp.IsZero() && !n.Spec.Unschedulable -> add to map and create replicas
4. default -> return reconcile.Result{}, nil

Reply:
Love it! Done.